linera-core/src/client.rs

```diff
@@ -516,28 +522,24 @@ where
                     > Some(height)
                 {
                     debug!("Accepting redundant notification for new message");
-                    return true;
                 }
                 if let Err(e) =
                     Self::find_received_certificates_from_validator(this.clone(), name, node).await
                 {
                     error!("Fail to process notification: {e}");
-                    return false;
                 }
                 if Self::local_next_block_height(this, origin.sender, &mut local_node).await
                     <= Some(height)
                 {
                     error!("Fail to synchronize new message after notification");
-                    return false;
                 }
             }
-            Reason::NewBlock { height, hash } => {
+            Reason::NewBlock { height, .. } => {
                 let chain_id = notification.chain_id;
                 if Self::local_next_block_height(this.clone(), chain_id, &mut local_node).await
                     > Some(height)
                 {
                     debug!("Accepting redundant notification for new block");
-                    return true;
                 }
                 match local_node
                     .try_synchronize_chain_state_from(name, node, chain_id)
```

Comment:
By the way, this may block the processing of other notifications for a long time if this validator is slow / malicious. Maybe create a TODO and a github task?

Reply:
Not sure what to do about it by the way: using a separate task/stream per validator is cleaner (and perhaps simplify the code) but it means we may download the same certificate twice. That is, until we find a way to interrupt redundant downloads early -- a mechanism which could probably be useful in other places.

Reply:
I created https://github.com/linera-io/linera-protocol/issues/1278 and https://github.com/linera-io/linera-protocol/issues/1279, but I'm also unsure what to do.

And this call is in `NewBlock`: While we don't have the chain's latest block anyway, is there even a point processing other notifications?

Reply:
We're talking about a malicious validator sending bad notifications to slow down the real ones